etcd/fs/fs_test.go

```diff
@@ -39,6 +43,31 @@ const (
 	superblock = "/some/superblock" // TODO: generate randomly per test?
 )
 
+func TestMain(m *testing.M) {
+	// run mgmt as etcd backend
+	cmdName, err := integration.BinaryPath()
+	if err != nil {
+		fmt.Printf("error getting binary path: %s", err)
+		os.Exit(1)
+	}
+	cmd := exec.Command(cmdName, "run", "--tmp-prefix")
+	if err := cmd.Start(); err != nil {
+		fmt.Printf("error starting command %v: %s", cmd, err)
+		os.Exit(1)
+	}
+
+	// cleanup when we're done
+	var exit int
+	defer func() {
+		cmd.Process.Signal(syscall.SIGQUIT)
+		cmd.Process.Kill()
+		os.Exit(exit)
+	}()
+
+	// run the tests
+	exit = m.Run()
```

Comment:
AFAIK setting exit this way (to be used in defer) is a bit of a dirty hack ;)

Reply:
There was more code in there before :) It's not neccessary as long as we ignore the errors.

Reply:
Oh, btw, can this patch also fix the FsTests that don't t.Errorf properly =D